src/plugins/intel_npu/src/plugin/npuw/llm_compiled_model.cpp

```diff
@@ -26,8 +28,58 @@ class TransposeValueTensors : public ov::pass::MatcherPass {
         using Ref = std::reference_wrapper<Context>;
     };
 
-    OPENVINO_MATCHER_PASS_RTTI("npuw::LLMCompiledModel::TransposeValueTensors");
-    TransposeValueTensors(Context::Ref ctx) {
+protected:
+    // generic part of matchers, to transpose v-tensors, and concat, and update matmul args 
+    void transpose_matmul_b(Context::Ref ctx, 
+            std::shared_ptr<ov::Node> node_param,
+            std::shared_ptr<ov::Node> node_concat,
+            std::shared_ptr<ov::Node> node_transpose,
+            std::shared_ptr<ov::Node> node_matmul) {
+
+        auto matched_param = std::static_pointer_cast<ov::op::v0::Parameter>(node_param);
+        auto matched_concat = std::static_pointer_cast<ov::op::v0::Concat>(node_concat);
+        auto matched_transpose = std::static_pointer_cast<ov::op::v1::Transpose>(node_transpose);
+        auto matched_matmul = std::static_pointer_cast<ov::op::v0::MatMul>(node_matmul);
+
+        auto param_shape = matched_param->get_partial_shape();
+        OPENVINO_ASSERT(param_shape.size() == 4u);
+        // NB: Transpose Parameter that correspond to V-tensor it will
+        // speed-up its multiplication with attention scores
+        std::swap(param_shape[2], param_shape[3]);
+        auto new_param = std::make_shared<ov::opset13::Parameter>(matched_param->get_element_type(), param_shape);
```

Comment:
minor:
do we need to create a new parameter here?
can we use "set_partial_shape" method of the old parameter?